torch/csrc/dynamo/eval_frame.c

```diff
@@ -476,14 +476,36 @@ inline static PyObject* eval_custom_code_impl(
   return result;
 }
 
+static const char* get_compile_context(PyThreadState* tstate) {
+    // Import the torch module
+    PyObject* torch_module = PyImport_ImportModule("torch._guards");
```

Comment:
Prefer having a global in C and defining builtin methods for accessing in Python, rather than having C accessing Python.

Reply:
That sounds like a reasonable solution as well. Some groups have stated that they would prefer that we don't cache graph values for use of profiling, as it forces us to update parameters in two locations, between python and C. However, since this affects performance, it sounds like it is a better route to take.